cmd/security-agent/subcommands/check/command.go

```diff
@@ -106,24 +107,16 @@ func commandsWrapped(bundleParamsFactory func() core.BundleParams) []*cobra.Comm
 }
 
 // RunCheck runs a check
-func RunCheck(log log.Component, config config.Component, checkArgs *CliParams) error {
+func RunCheck(log log.Component, config config.Component, statsd statsd.Component, checkArgs *CliParams) error {
 	hname, err := hostname.Get(context.TODO())
 	if err != nil {
 		return err
 	}
 
-	var statsdClient *ddgostatsd.Client
+	var statsdClient ddgostatsd.ClientInterface
 	metricsEnabled := config.GetBool("compliance_config.metrics.enabled")
 	if metricsEnabled {
-		// Create a statsd Client
-		statsdAddr := os.Getenv("STATSD_URL")
-		if statsdAddr == "" {
-			// Retrieve statsd host and port from the datadog agent configuration file
-			statsdHost := pkgconfig.GetBindHost()
-			statsdPort := config.GetInt("dogstatsd_port")
-			statsdAddr = fmt.Sprintf("%s:%d", statsdHost, statsdPort)
-		}
-		cl, err := ddgostatsd.New(statsdAddr)
+		cl, err := statsd.GetForHostPort(pkgconfig.GetBindHost(), config.GetInt("dogstatsd_port"))
```

Comment:
What about the env var `STATSD_URL` ? It seems we do not respect it with this change. What am I missing ? (To be clear I'm not sure if we should respect it or not)

Reply:
https://github.com/DataDog/datadog-agent/pull/20613/files#diff-0d99624b4de423e76754bedadba13ba6040f055f9eb771dac1cc3ad68e610466R50

STATSD_URL still is  used if available, as it should be. The host/port thing is a fallback guessing the config from the agent config (and it doesn't support unix sockets)